lib/scope.js

```diff
@@ -633,167 +623,213 @@ module.exports = {
+          if ( !uses_proxies ) {  // If this gets set to `true` once, leave it alone
+            // x, i, j, etc. See notes below about the reasoning
+            // AVOID `.test()` - it advances through a string each time you call it on the
+            // string, so you can have unexpected results.
+            uses_proxies = guess.var.match( proxies_regex );  // 
           }
+          continue;
+        }
 
-        }  // ends for each table row
-      }  // ends for each field row possibility
-      matches_by_field.push( match );
-    }  // ends for each field
-
-    // Give the developer some feedback if needed
-    for ( let field of matches_by_field ) {
-      if ( field.matching_table_rows.length > 1 ) {
-        let msg = `Warning: there are multiple matches to a field. The story table row that appears first will be used. Here is the data: ${JSON.stringify( field )}`
-        await scope.addToReport(scope, { key: 'ids', value: msg });
+      }  // ends for each table row
+    }  // ends for each field row possibility
+
+    // Some fields will have multiple matches from the story table.
+    // The only reason we would care about multiple story table row
+    // matches is if there is a possible proxy var conflict.
+    // When a dev has a loop with an index var or generic object,
+    // the same variable name will be used over and over, e.g. `users[i]`.
+    // That's what will be in the story table, so there will be multiple
+    // matches for a field setting `users[i]`. For this program to
+    // know which table row to use on which page, we need the one unique var that
+    // will appear on the page - the sought var. When a row with `users[i]`
+    // has a sought var that matches the page sought var, the
+    // program knows to use that row to set the value for that page.
+
+    // If needed, only get the matches that also match the sought var
+    // For now, we're being flexible about it - if we don't think we
+    // need that sought var, we don't prevent data getting through.
+    // e.g. If an interview has a list of `users`, but they only set
+    // data for `users[0]`, there's no need to check the sought var.
+    // That way devs with simple cases can ignore the sought var column.
+
+    // If field uses proxies and there has been more than one match
+    // we now use the trigger var to narrow down the matches.
+    if ( uses_proxies && matches.length > 1 ) {
+      // Multiple matches means this came from a story
+
+      // Feedback for devs - we need the trigger var but we don't have it (field data)
+      // TODO: Add docs and link to docs on how to add the trigger var
+      if ( field.trigger === undefined ) { await scope.addToReport(scope, { key: 'ids', value: `WARNING: A field on this page uses an index variable or a generic object, but the page does not contain an element with the trigger variable's name. Tell the developer to add an element to the page with the id "sought_variable"` }); }
+
```

Comment:
If `field.trigger` is undefined, stop here. This probably wouldn't be expected behavior for the developer for this to continue and not collect matches.